### PR TITLE
Bugfix: Various issues with edge index

### PR DESCRIPTION
**Pull request by andrejtonev**
EdgeType index would assign every element as obsolete
EdgeTypeProperty index did not run GC
Allowed edge index creation even when properties on edges was false
Recovered edge indices even if properties on edges was false
EdgeType was not updated via `CreateEdgeEx`

### CODE REVIEW

**Comment by andrejtonev** [comment]
Git commit description, explain the changes you made here

---
__*Leave above in PR description, copy the below into a comment*__
___


### Tracking
- [ ] **[Link to Epic/Issue]**


### Standard development
- [ ] Update unit/E2E tests
- [ ] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch this branch


### CI Testing Labels
- [ ] Select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_


### Documentation checklist
- [ ] Add the documentation label
- [ ] Add the bug / feature label
- [ ] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [ ] Write a release note, including added/changed clauses
    - **[ Release note text ]**
- [ ] **[ Documentation PR link memgraph/documentation#XXXX ]**
    - [ ] Is back linked to this development PR
- [ ] **[ Tag someone from docs team ]**

File: src/dbms/inmemory/replication_handlers.cpp [hunk 1]
```diff
@@ -303,9 +303,9 @@ void InMemoryReplicationHandlers::SnapshotHandler(dbms::DbmsHandler *dbms_handle
     storage->wal_seq_num_ = 0;
 
     spdlog::trace("Recovering indices and constraints from snapshot.");
-    memgraph::storage::durability::RecoverIndicesAndStats(recovered_snapshot.indices_constraints.indices,
-                                                          &storage->indices_, &storage->vertices_,
-                                                          storage->name_id_mapper_.get());
+    memgraph::storage::durability::RecoverIndicesAndStats(
+        recovered_snapshot.indices_constraints.indices, &storage->indices_, &storage->vertices_,
+        storage->name_id_mapper_.get(), storage->config_.salient.items.properties_on_edges);
```
Comment (as51340) [inline src/dbms/inmemory/replication_handlers.cpp:308]: getter for properties_on_edges?

File: src/query/interpreter.cpp [hunk 2]
```diff
@@ -2930,6 +2930,10 @@ PreparedQuery PrepareEdgeIndexQuery(ParsedQuery parsed_query, bool in_explicit_t
     throw IndexInMulticommandTxException();
   }
 
+  if (!current_db.db_acc_->get()->config().salient.items.properties_on_edges) {
```
Comment (as51340) [inline src/query/interpreter.cpp:2933]: same, use getter

File: src/storage/v2/durability/durability.cpp [hunk 3]
```diff
@@ -421,7 +426,7 @@ std::optional<RecoveryInfo> Recovery::RecoverData(
     if (!utils::DirExists(wal_directory_)) {
       // Apply data dependant meta structures now after all graph data has been loaded
       RecoverIndicesStatsAndConstraints(vertices, name_id_mapper, indices, constraints, config, recovery_info,
-                                        indices_constraints);
+                                        indices_constraints, config.salient.items.properties_on_edges);
```
Comment (as51340) [inline src/storage/v2/durability/durability.cpp:429]: same

**Review by as51340 — APPROVED** [review]
LGTM

File: src/query/interpreter.cpp (hunk 2 above)
Comment (andrejtonev) [reply src/query/interpreter.cpp:2933]: same answer
tldr out of scope

File: src/storage/v2/durability/durability.cpp (hunk 3 above)
Comment (andrejtonev) [reply src/storage/v2/durability/durability.cpp:429]: same answer
tldr out of scope
